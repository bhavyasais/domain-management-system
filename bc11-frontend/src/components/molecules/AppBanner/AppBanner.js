import React from "react";
import Typography from "@material-ui/core/Typography";
import { makeStyles } from "@material-ui/core/styles";
import Image from "react-image-resizer";
import banner from "../../../assets/banner.svg";
import ButtonAtom from "../../atoms/Button/ButtonAtom";
import {DialogTemplate} from '../../templates/DialogTemplate/CollabCardsDialogTemplate';
import { useHistory } from 'react-router-dom';

const useStyles = makeStyles(() => ({
  root: {},
  topmargin: {
    marginTop: "30px",
    color: "grey",
  },
  buttonmargin: {
    marginTop: "10px",
  },
}));

export default function AppBanner(props) {
  const classes = useStyles();
  const history = useHistory();
  const [open, setOpen] = React.useState(false);

  const handleClickOpen = () => {
    setOpen(true);
  };
  const handleClose = () => {
    setOpen(false);
  };
  const handleOnClick=()=>{
    history.push("/base");
  }
  return (
    <div className={props && props.childClasses && props.childClasses.root ? props.childClasses.root : classes.root}>
      <div align="center">
        <Image
          src={banner}
          height={props && props.height ? props.height: 400}
          width={props && props.width ? props.width : 500}
          className={props && props.childClasses && props.childClasses.alignCenter ? props.childClasses.alignCenter : classes.alignCenter}
        />
      </div>
      <Typography variant="h5" align="center" className={props && props.childClasses && props.childClasses.topmargin ? props.childClasses.topmargin : classes.topmargin}>
        Your active permissions will appear here.
        <div>
          <ButtonAtom
            variant="contained"
            value="Enable Access"
            color="#1d4cd7"
            className={props && props.childClasses && props.childClasses.buttonmargin ? props.childClasses.buttonmargin : classes.buttonmargin}
            onClick={handleClickOpen}
          />
          <DialogTemplate handleClose={handleClose} open={open} onClick={handleOnClick}/>
        </div>
      </Typography>
    </div>
  );
}