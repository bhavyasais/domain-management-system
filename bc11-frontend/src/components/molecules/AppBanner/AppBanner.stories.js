import React from "react";
import AppBanner from "./AppBanner";

export default {
  title: "Molecules/Banner",
};

export const Banner = () => {
  return <AppBanner />;
};
