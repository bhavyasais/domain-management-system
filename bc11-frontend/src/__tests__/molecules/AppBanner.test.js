import AppBanner from "../../components/molecules/AppBanner/AppBanner";
import React from "react";
import renderer from "react-test-renderer";
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';

describe("Banner", () => {
  let component;
  const handleClick = jest.fn();
  beforeEach(() => {
    act(() => {
      component = mount(<AppBanner />);
    });
  });
  it("Renders correctly", () => {
    let wrapper = renderer.create(<AppBanner />);
    expect(wrapper).toMatchSnapshot();
  });

  it("opens the modal", () => {
    const wrapper = mount(<AppBanner onClick={handleClick}/>);
    wrapper.find('ButtonAtom').at(0).props().onClick();
    wrapper.find('DialogTemplate').at(0).props().handleClose();
  });

});
