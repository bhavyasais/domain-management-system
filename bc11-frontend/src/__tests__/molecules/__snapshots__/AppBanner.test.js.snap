// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    align="center"
  >
    <div
      style={
        Object {
          "backgroundColor": undefined,
          "height": 400,
          "position": "relative",
          "width": 500,
        }
      }
    >
      <img
        onError={[Function]}
        onLoad={[Function]}
        src="banner.svg"
        style={
          Object {
            "display": "block",
            "height": 0,
            "left": 250,
            "position": "absolute",
            "top": 200,
            "width": 0,
          }
        }
      />
    </div>
  </div>
  <h5
    className="MuiTypography-root makeStyles-topmargin-2 MuiTypography-h5 MuiTypography-alignCenter"
  >
    Your active permissions will appear here.
    <div>
      <div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-buttonmargin-3"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "backgroundColor": "#1d4cd7",
              "maxWidth": undefined,
            }
          }
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label makeStyles-buttonLabel-4"
          >
            <p
              className="MuiTypography-root makeStyles-buttonLabel-4 MuiTypography-body1"
            >
              Enable Access
            </p>
          </span>
        </button>
      </div>
      <div
        className="makeStyles-root-5"
      />
    </div>
  </h5>
</div>
`;
